### PR TITLE
Fix typos (`the the` > `the`)

### PR DESCRIPTION
**Pull request by eliasp**
It looks like humans aren't only very bad at recognizing a double `the` in a sentence when reading [1], but they're also very good at sneaking them in unintentionally without noticing.

Fix this by running:
```
ack --ignore-dir=doc/man --ignore-dir=doc/topics/releases -i 'the the\W' -l | xargs sed -i 's|the the\(\W\)|the\1|g'
```

[1] Eye movements and word skipping during reading: Effects of word length and predictability: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3543826/

### What does this PR do?
Fixes a common typo

### What issues does this PR fix or reference?
None

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by Ch3LL — DISMISSED** [review]

**Review by s0undt3ch — DISMISSED** [review]

**Review by dwoz — DISMISSED** [review]

**Comment by waynew** [comment]
Hey @eliasp looks like we missed merging this a while ago. Do you mind rebasing this on `master`?

Might be worth just running this against master to see if there are *more* double `the`'s there :trollface:

**Comment by eliasp** [comment]
Rebased on `master` + re-created the commit by running it again.

**Comment by eliasp** [comment]
Failing CI is completely unrelated (and only editing doc/comments shouldn't have any influence on CI anyways 😄)

**Review by waynew — DISMISSED** [review]

**Comment by eliasp** [comment]
Rebased against latest master to get some merge conflicts ouf of the way

**Comment by Ch3LL** [comment]
looks like there are some failure that need to be resolved here. Looks like the rebase might have caused some issues. We recently added pre-commit which should run black against your code. I noticed that the PR has some single ticks where black requires double quotes.

File: doc/topics/development/tests/unit.rst [hunk 1]
```diff
@@ -174,8 +174,13 @@ What happens in the above example is we mock a call to
...
 available, since that's not actually part of what's being tested, we mocked that
-import by patching ``sys.modules`` when tests are running.
-
+import ``sys.modules``
+import are
+import by
+import patching
+import running.
+import tests
+import when
```
Comment (s0undt3ch) [inline doc/topics/development/tests/unit.rst:183]: All these imports.... :grin:

File: doc/topics/tutorials/firewall.rst [hunk 2]
```diff
@@ -207,7 +207,7 @@ Whitelist communication to Master
 =================================
 
 There are situations where you want to selectively allow Minion traffic
-from specific hosts or networks into your Salt Master. The first
```
Comment (s0undt3ch) [inline doc/topics/tutorials/firewall.rst:210]: Unintentional delete?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: doc/topics/tutorials/firewall.rst (hunk 2 above)
Comment (waynew) [reply doc/topics/tutorials/firewall.rst]: fixed

File: doc/topics/development/tests/unit.rst (hunk 1 above)
Comment (waynew) [reply doc/topics/development/tests/unit.rst]: :eyes: 

:fire:

**Review by waynew — APPROVED** [review]
🎉

**Comment by waynew** [comment]
@s0undt3ch are you good here?

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**